Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -59,6 +59,24 @@ struct SymbolGraphLoader {
         let bundle = self.bundle
         let dataProvider = self.dataProvider
         
+        /// Computes the default availbiality based on the `inheritDefaultAvailability` option.
+        let defaultAvailabilities: ([DefaultAvailability.ModuleAvailability]?) -> [DefaultAvailability.ModuleAvailability]? = { defautAvailabilities in
```

Comment:
Why does this need to be a closure? I can't see a requirement for this.

Reply:
Now that you mention it; why is the symbol graph loader responsible for modifying the decoded Info.plist values?

Reply:
> Now that you mention it; why is the symbol graph loader responsible for modifying the decoded Info.plist values?

Is during the symbol graph loading that each symbol gets assigned the default availability data. I could create an intermediary structure to not modify the input data, would that be better? CC: @d-ronnqvist

Reply:
Unless I'm missing some reason why it wouldn't be possible;
I think it would be better to create the `.available(version: nil)` values when decoding the Info.plist so that they don't need to be modified after the fact.

Reply:
> I think it would be better to create the .available(version: nil) values when decoding the Info.plist so that they don't need to be modified after the fact.

The framework availability information must remain, so this change only applies at the symbol level. Modifying the default availability information when decoding it would remove the availability at the top level page.

Reply:
> > I think it would be better to create the .available(version: nil) values when decoding the Info.plist so that they don't need to be modified after the fact.
> 
> 
> 
> The framework availability information must remain. Modifying the default availability information when decoding it would remove the availability at the top level page. 

To play devil's advocate for a second:
We don't really have a backwards compatibility issue here because the developer needs to specify a new configuration key to opt in to the new behavior. 

From one perspective it might be preferable to display the version information on the module page but not the individual symbols' pages, but from another perspective it's more consistent if the configuration applies equally to all pages. At the very least "must" is a much to strong assertion. 

If the developer wants to display version information on the module page they can accomplish that by explicitly specifying introduced versions in `@Available` directives in the module page's documentation extension file.

Reply:
More seriously; 
If this is intentionally deferring a modification until after a certain read-access in order to get two different behaviors out of the same property then I don't quite like that. 

If we want to different behaviors then I would prefer if we made that behavioral difference explicit. 

One way to accomplish that could be to move access behind a function with a parameter. Another way to accomplish that could be to introduce a second property for the other behavior.